streams/src/main/java/org/apache/kafka/streams/kstream/Windows.java

```diff
@@ -40,7 +40,11 @@
 
     // By default grace period is 24 hours for all windows,
     // in other words we allow out-of-order data for up to a day
-    protected static final long DEFAULT_GRACE_PERIOD_MS = 24 * 60 * 60 * 1000L;
+    // This behavior is now deprecated
+    protected static final long DEPRECATED_OLD_24_HR_GRACE_PERIOD = 24 * 60 * 60 * 1000L;
```

Comment:
This naming is confusing to uses. This will make user think that 24 hours grace period cannot be used anymore. No I don't think that's what we want. 24 hours is still good to use, if user believe that's what they want. That means, we don't **deprecate** the 24 hours grace period, just don't set as default value, so we should not name it as "deprecated", "old", things.

Correct me if I'm wrong, @izzyacademy @ableegoldman . Thank you

Reply:
This is not a publicly available constant. It is only used internally by the implementers so I think it is OK. The reasoning for deprecating the constant is explained thoroughly in the KIP so I think they will understand. I can add more details to the Java comments to clarify. But thanks @showuon  for bring it up in the review.

Reply:
Yeah it's an internal config so I hope they wouldn't assume anything from the name and extrapolate to what they can and can't use. That said, it does appear in these classes which are public themselves, so users are still going to see it. But the important thing is that it makes sense to us, the devs, who will actually be using it -- I think `DEPRECATED_DEFAULT_24_HR_GRACE_PERIOD` is a bit more clear, just need to sneak the word "default" in there somewhere imo